data/modules/Scoop/Scoop.lua

```diff
@@ -0,0 +1,837 @@
+			mission.pirate = nil
+			removeMission(mission)
+			pirate:HyperjumpTo(nearbySystem())
+		end
+	end)
+end
+
+local isQualifiedFor = function(reputation, ad)
+	return reputation > (ad.reward/100) or false
+end
+
+local onDelete = function (ref)
+	ads[ref] = nil
+end
+
+local isEnabled = function (ref)
+	return ads[ref] ~= nil and isQualifiedFor(Character.persistent.player.reputation, ads[ref])
+end
+
+local onChat = function (form, ref, option)
+	local ad = ads[ref]
+	local player = Game.player
+	local debris, ship, radius, mindist, maxdist
+
+	form:Clear()
+
+	if option == -1 then
+		form:Close()
+		return
+	end
+
+	local qualified = isQualifiedFor(Character.persistent.player.reputation, ad)
+
+	form:SetFace(ad.client)
+
+	if not qualified then
+		form:SetMessage(l["DENY_" .. Engine.rand:Integer(1, getNumberOfFlavours("DENY"))])
+		return
+	end
+
+	form:AddNavButton(ad.planet)
+
+	if option == 0 then
+		local introtext = string.interp(ad.introtext, {
+			client = ad.client.name,
+			shipid = ad.pirate_label,
+			star   = ad.star:GetSystemBody().name,
+			planet = ad.planet:GetSystemBody().name,
+			cash   = Format.Money(math.abs(ad.reward), false),
+		})
+		form:SetMessage(introtext)
+
+	elseif option == 1 then
+		form:SetMessage(l["WHY_NOT_YOURSELF_" .. ad.id])
+
+	elseif option == 2 then
+		form:SetMessage(string.interp(l["HOW_MUCH_TIME_" .. ad.id], { star = ad.star:GetSystemBody().name, date = Format.Date(ad.due) }))
+
+	elseif option == 3 then
+		if ad.reward > 0 and player:CountEquip(Equipment.misc.cargo_scoop) == 0 and player:CountEquip(Equipment.misc.multi_scoop) == 0 then
+			form:SetMessage(l.YOU_DO_NOT_HAVE_A_SCOOP)
+			form:RemoveNavButton()
+			return
+		end
+
+		if ad.reward < 0 and player:GetMoney() < math.abs(ad.reward) then
+			form:SetMessage(l.YOU_DO_NOT_HAVE_ENOUGH_MONEY)
+			form:RemoveNavButton()
+			return
+		end
+
+		form:RemoveAdvertOnClose()
+		ads[ref] = nil
+
+		radius = ad.planet:GetSystemBody().radius
+		if ad.planet:GetSystemBody().superType == "ROCKY_PLANET" then
+			mindist = radius * 2.5
+			maxdist = radius * 3.5
+		else
+			mindist = radius * 25
+			maxdist = radius * 35
+		end
+		debris = spawnDebris(ad.debris_type, ad.amount, ad.planet, mindist, maxdist, ad.due - Game.time)
+
+		if ad.reward < 0 then player:AddMoney(ad.reward) end
+		if ad.deliver_to_ship then
+			ship = spawnPirate(ad.star, ad.pirate_label)
+		end
+
+		local mission = {
+			type              = "Scoop",
+			location          = ad.location,
+			introtext         = ad.introtext,
+			client            = ad.client,
+			station           = ad.station.path,
+			star              = ad.star,
+			planet            = ad.planet,
+			id                = ad.id,
+			debris            = debris,
+			amount            = #debris,
+			reward            = ad.reward,
+			due               = ad.due,
+			return_to_station = ad.return_to_station,
+			deliver_to_ship   = ad.deliver_to_ship,
+			pirate            = ship,
+			pirate_label      = ad.pirate_label,
```

Comment:
Unless I'm horribly missing something, the ship you deliver goods to isn't actually a pirate (doesn't attack you, stumble when he walks, or smell like rum) and I don't think anything in the Mission Support Framework that is our ad-hoc class setup actually needs this ship to be referred to with a variable named `pirate`.

It's a bit of a papercut, but calling this ship something more descriptive would greatly ease the maintenance burden of this module!